Clustering/hybridEcalSplitter/src/DDStripSplitter.cc

```diff
@@ -0,0 +1,790 @@
+      h_staveZ[i]  = new TH2F("staveZ_"+reg,  "staveZ_"+reg,  100,-5000,5000,10,0,10);
+      h_moduleX[i] = new TH2F("moduleX_"+reg, "moduleX_"+reg, 100,-2000,2000,10,0,10);
+      h_moduleY[i] = new TH2F("moduleY_"+reg, "moduleY_"+reg, 100,-2000,2000,10,0,10);
+      h_moduleZ[i] = new TH2F("moduleZ_"+reg, "moduleZ_"+reg, 100,-5000,5000,10,0,10);
+    }
+    
+    for (int ieb=0; ieb<2; ieb++) {
+      for (int is=0; is<10; is++) {
+	for (int im=0; im<10; im++) {
+	  TString hhname = ieb==0 ? "barrel" : "endcap";
+	  hhname+="_stave"; hhname+=is;
+	  hhname+="_module"; hhname+=im;
+	  h_cth_phi[ieb][is][im] = new TH2F(hhname+"_angle",hhname+"_angle",100,-1,1,100,-TMath::Pi(), TMath::Pi());
+	  h_XY[ieb][is][im] = new TH2F(hhname+"_xy",hhname+"_xy",100,-3000,3000,100,-3000,3000);
+	}
+      }
+    }
+  }
+
+  return;
+}
+
+void DDStripSplitter::processRunHeader( LCRunHeader*  /*run*/) { 
+   return;
+}
+
+
+
+void DDStripSplitter::setupGeometry() {
+
+
+  unsigned int includeFlag(0);
+  unsigned int excludeFlag(0);
+
+  includeFlag = ( dd4hep::DetType::CALORIMETER | dd4hep::DetType::ELECTROMAGNETIC | dd4hep::DetType::BARREL);
+  excludeFlag = ( dd4hep::DetType::AUXILIARY | dd4hep::DetType::FORWARD );
+
+  dd4hep::Detector & lcdd = dd4hep::Detector::getInstance();
+  const std::vector< dd4hep::DetElement>& theDetectors = dd4hep::DetectorSelector(lcdd).detectors(  includeFlag, excludeFlag ) ;
+  streamlog_out( DEBUG2 ) << " getExtension :  includeFlag: " << dd4hep::DetType( includeFlag ) << " excludeFlag: " << dd4hep::DetType( excludeFlag )
+			  << "  found : " << theDetectors.size() << "  - first det: " << theDetectors.at(0).name() << std::endl ;
+
+  if( theDetectors.size()  != 1 ){
+    std::stringstream es ;
+    streamlog_out (ERROR) << " getExtension: selection is not unique (or empty)  includeFlag: " << dd4hep::DetType( includeFlag ) << " excludeFlag: " << dd4hep::DetType( excludeFlag )
+			  << " --- found detectors : " ;
+    for( unsigned i=0, N= theDetectors.size(); i<N ; ++i ){
+      streamlog_out (ERROR) << theDetectors.at(i).name() << ", " ;
+    }
+    assert(0);
+  }
+
+  int ical=0;
+  _caloGeomData = theDetectors.at(ical).extension<dd4hep::rec::LayeredCalorimeterData>();
+  if ( ! _caloGeomData ) {
+    streamlog_out ( WARNING ) << "could not get calorimeter geometry information!" << endl;
+    assert(0);
+  }
+  _symmetry = _caloGeomData->inner_symmetry;
+  _stripLength = _caloGeomData->layers[0].cellSize0;
+  _stripWidth  = _caloGeomData->layers[0].cellSize1;
+
+  // check that layer 1 is orthogonal to this one
+  float lay1_strSize0 = _caloGeomData->layers[1].cellSize0;
+  float lay1_strSize1 = _caloGeomData->layers[1].cellSize1;
+  if ( fabs(lay1_strSize0-_stripWidth)/_stripWidth > 0.05 || fabs(lay1_strSize1-_stripLength)/_stripLength > 0.05 ) {
+    cout << "doesn't look like a basic strip-based calo?? don't know how to deal with this geometry!" << endl;
```

Comment:
Can you replace all `cout` calls by `streamlog_out()` calls ?

Reply:
done